File: src/model/item/Vehicle.java
package model.item;

import java.util.List;

import core.ConstantStore;

import model.Condition;
import model.Inventoried;
import model.Inventory;
import model.Item;

/**
 * A vehicle is anything that is classified as a vehicle
 */
public class Vehicle extends Item implements Inventoried{

	private final Inventory cargo;
	
	private final int MAX_INVENTORY_SIZE;
	
	private final double MAX_INVENTORY_WEIGHT;

	/**
	 * Makes a vehicle to be used to get to oregon
	 * @param type The type of vehicle
	 */
	public Vehicle(double maxWeight, ItemType type) {
		super(type);
		this.MAX_INVENTORY_SIZE = Integer.parseInt(ConstantStore.get("ITEMS", type + "_MAX_INV_SIZE"));
		this.MAX_INVENTORY_WEIGHT = Double.parseDouble(ConstantStore.get("ITEMS", type + "_MAX_INV_WEIGHT"));
		this.cargo = new Inventory(MAX_INVENTORY_SIZE, MAX_INVENTORY_WEIGHT);
	}
	
	@Override
	public Condition getStatus() {
		return status.copy();
	}
	
	@Override
	public void increaseStatus(int amount) {
		status.increase(amount);
	}
	
	@Override
	public void decreaseStatus(int amount) {
		status.decrease(amount);
	}
	
	@Override
	public Inventory getInventory() {
		return cargo;
	}
	
	@Override
	public void addItemsToInventory(List<Item> items) {
		cargo.addItemsToInventory(items);
	}
	
	@Override
	public void addItemToInventory(Item item) {
		cargo.addItemToInventory(item);
	}
	
	@Override
	public List<Item> removeItemFromInventory(ItemType itemIndex, int quantity) {
		return cargo.removeItemFromInventory(itemIndex, quantity);
	}
	
	@Override
	public boolean canGetItem(ItemType itemType, int numberOf) {
		return cargo.canGetItems(itemType, numberOf);
	}
	
	@Override
	public int getMaxSize() {
		return MAX_INVENTORY_SIZE;
	}
	
	@Override
	public double getMaxWeight() {
		return MAX_INVENTORY_WEIGHT;
	}
	
	@Override
	public double getWeight() {
		return cargo.getWeight();
	}

	public void repair(int amount) {
		status.increase(amount);
	}
}

File: web/src/model/Vehicle.ts
import { Item } from "./Item";
import { Inventory } from "./Inventory";
import { ItemType } from "./ItemType";
import { Condition } from "./Condition";
import { ConstantStore } from "../core/ConstantStore";

export class Vehicle extends Item {
  private readonly cargo: Inventory;
  private readonly MAX_INVENTORY_SIZE: number;
  private readonly MAX_INVENTORY_WEIGHT: number;

  constructor(type: ItemType) {
    super(type);
    this.MAX_INVENTORY_SIZE = parseInt(
      ConstantStore.get("ITEMS", `${type}_MAX_INV_SIZE`),
      10,
    );
    this.MAX_INVENTORY_WEIGHT = parseFloat(
      ConstantStore.get("ITEMS", `${type}_MAX_INV_WEIGHT`),
    );
    this.cargo = new Inventory(
      this.MAX_INVENTORY_SIZE,
      this.MAX_INVENTORY_WEIGHT,
    );
  }

  getStatus(): Condition {
    return this.getCondition().copy();
  }

  increaseStatus(amount: number): void {
    this.getCondition().increase(amount);
  }

  decreaseStatus(amount: number): void {
    this.getCondition().decrease(amount);
  }

  getInventory(): Inventory {
    return this.cargo;
  }

  addItemsToInventory(items: Item[]): void {
    this.cargo.addItemsToInventory(items);
  }

  addItemToInventory(item: Item): void {
    this.cargo.addItemToInventory(item);
  }

  removeItemFromInventory(itemType: ItemType, quantity: number): Item[] | null {
    return this.cargo.removeItemFromInventory(itemType, quantity);
  }

  canGetItem(itemType: ItemType, numberOf: number): boolean {
    return this.cargo.canGetItems(itemType, numberOf);
  }

  getMaxSize(): number {
    return this.MAX_INVENTORY_SIZE;
  }

  getMaxWeight(): number {
    return this.MAX_INVENTORY_WEIGHT;
  }

  getWeight(): number {
    return this.cargo.getWeight();
  }

  repair(amount: number): void {
    this.getCondition().increase(amount);
  }
}
